ldap/servers/plugins/memberof/memberof.c

```diff
@@ -2820,9 +2833,16 @@ memberof_fixup_task_thread(void *arg)
     if (!task) {
         return; /* no task */
     }
+
+    PR_Lock(fixup_lock);
+    fixup_running = PR_TRUE;
```

Comment:
Only 'fixup_running' looks protected by the lock. I would unlock just after the setting of fixup_running to make it clear that the others fixup_* are not protected by the lock.

Reply:
only "fixup_running" needs to be protected